python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -731,12 +731,18 @@ def conditions_for_candidate(
             )
         ):
             conditions.add(MissingAutoMaterializeCondition())
-
-        # if the parent has been updated
-        if auto_materialize_policy.on_new_parent_data and not instance_queryer.is_reconciled(
-            asset_partition=candidate
-        ):
-            conditions.add(ParentMaterializedAutoMaterializeCondition())
+        else:
+            # if the parent has been updated
+            if auto_materialize_policy.on_new_parent_data:
+                ancestors = instance_queryer.get_ancestors_missing_or_with_unreconciled_children(
+                    candidate
+                )
+                if ancestors:
+                    conditions.add(
+                        ParentMaterializedAutoMaterializeCondition(
```

Comment:
Currently there's two cases we're conflating: parent materialized, and parent is missing and will be materialized (e.g. if the parent is time partitioned and got a new partition), so we're materializing this asset with it. I don't think it's urgent but we can separate those cases eventually